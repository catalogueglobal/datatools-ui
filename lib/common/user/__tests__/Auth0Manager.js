import auth0 from '../Auth0Manager'

describe('common > user > Auth0Manager >', () => {
  it('should login with lock', (done) => {
    // setup localStorage
    const storage = {}
    window.localStorage = {
      getItem: (k) => storage[k],
      setItem: (k, v) => { storage[k] = v }
    }

    // setup mock action
    const receiveTokenAndProfile = jest.fn()

    // set auth0 lock options to generate test response
    auth0.lockOptions = {
      fakeAuthenticatedToken: 'fake-token',
      getProfileSuccess: true,
      getProfileResult: {
        app_metadata: {
          datatools: []
        }
      }
    }

    auth0.loginWithLock({
      receiveTokenAndProfile
    })

    // do a timeout because the promise needs to execute
    setTimeout(() => {
      expect(receiveTokenAndProfile.mock.calls).toMatchSnapshot()
      done()
    }, 10)
  })
})
