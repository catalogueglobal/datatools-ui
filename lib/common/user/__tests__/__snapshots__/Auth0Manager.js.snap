// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common > user > Auth0Manager > should login with lock 1`] = `
Array [
  Array [
    Object {
      "profile": Object {
        "app_metadata": Object {
          "datatools": Array [],
        },
      },
      "token": "fake-token",
    },
  ],
]
`;
